docs/server-event-types.md

```diff
@@ -0,0 +1,49 @@
+---
+id: server-event-types
+title: Event type reference
+sidebar_label: Event types
+---
+
+Event types are defined in the Server API [event_type.proto](https://github.com/temporalio/api/blob/master/temporal/api/enums/v1/event_type.proto) file.
+
+- **EVENT_TYPE_UNSPECIFIED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_STARTED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_FAILED**:
+- **EVENT_TYPE_WORKFLOW_EXECUTION_TIMED_OUT**:
+- **EVENT_TYPE_WORKFLOW_TASK_SCHEDULED**:
+- **EVENT_TYPE_WORKFLOW_TASK_STARTED**:
+- **EVENT_TYPE_WORKFLOW_TASK_COMPLETED**:
+- **EVENT_TYPE_WORKFLOW_TASK_TIMED_OUT**:
+- **EVENT_TYPE_WORKFLOW_TASK_FAILED**:
+- **EVENT_TYPE_ACTIVITY_TASK_SCHEDULED**:
+- **EVENT_TYPE_ACTIVITY_TASK_STARTED**:
+- **EVENT_TYPE_ACTIVITY_TASK_COMPLETED**:
+- **EVENT_TYPE_ACTIVITY_TASK_FAILED**:
+- **EVENT_TYPE_ACTIVITY_TASK_TIMED_OUT**:
+- **EVENT_TYPE_ACTIVITY_TASK_CANCEL_REQUESTED**:
+- **EVENT_TYPE_ACTIVITY_TASK_CANCELED**:
```

Comment:
- **EVENT_TYPE_ACTIVITY_TASK_SCHEDULED**:
This event type indicates the activity task scheduled, meaning SDK should pick up this activity task and execute.
This event type contains activity inputs, as well as activity timeout configurations.

- **EVENT_TYPE_ACTIVITY_TASK_STARTED**:
This event type indicates the activity task started, meaning SDK picked up the activity task, and is processing this activity invocation.

- **EVENT_TYPE_ACTIVITY_TASK_COMPLETED**:
This event type indicates the activity task completed, meaning SDK picked up the activity task, and successfully completed the activity task.
This event type contains activity execution result.

- **EVENT_TYPE_ACTIVITY_TASK_FAILED**:
This event type indicates the activity task completed, meaning SDK picked up the activity task, and unsuccessfully completed the activity task.
This event type contains activity execution error.

- **EVENT_TYPE_ACTIVITY_TASK_TIMED_OUT**:
This event type indicates the activity is timed out by Temporal server, due to activity not completed within timeout settings.

- **EVENT_TYPE_ACTIVITY_TASK_CANCEL_REQUESTED**:
This event type indicates the activity should be cancelled. SDK will know an activity should be cancelled during activity heartbeat. 

- **EVENT_TYPE_ACTIVITY_TASK_CANCELED**:
This event type indicates the activity is cancelled.